### PR TITLE
Refactoring of codegen-py jinja context and type/variable naming rules

### PR DESCRIPTION
**Pull request by lreiher**
- simplify naming rules of generated types and member fields; no more special handling of certain strings
- heavily refactor Jinja context of codegen-py to make it more understandable and maintainable
- make curly brace usage in Jinja templates consistent
- related to #45

### Breaking Changes due to naming simplifications

See 7d939ecdbbbb56492ef37b2093c88bf2a7c0f239.

### CODE REVIEW

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 1]
```diff
@@ -44,135 +44,113 @@
...
-        is_const (bool): whether the field is a constant
+    # remove leading _, replace - with _, remove double _
+    ss = ss.lstrip("_").replace("-", "_").replace("__", "_")
 
-    Returns:
-        str: a_ros_message_field or A_ROS_MESSAGE_CONSTANT
-    """
+    # TODO: use this or not?
+    # add _ before number following lower-case letter
+    # ss = re.sub("([a-z])([0-9])", r"\1_\2", ss)
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:105]: TODO

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 2]
```diff
@@ -44,135 +44,113 @@
...
-        s (str): e.g., "class"
+        s (str): A-Message-Type
+
+    Returns:
+        str: a_message_type
+    """
+
+    # TODO: only works if "add _ before number following lower-case letter" is not used
+    # TODO: i think it leads to include problems, e.g., "#include <etsi_its_mapem_ts_msgs/msg/node_xy_20b.hpp>" in mapem_ts_conversion/convertNodeXY20b.h
+    return validRosField(s)
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:131]: TODO

File: utils/codegen/codegen-py/templates/convertSequenceType.h.jinja2 [hunk 3]
```diff
@@ -26,85 +26,85 @@ SOFTWARE.
 
 #pragma once
 
-#include <etsi_its_{{ etsi_type }}_coding/{{ etsi_type }}_{{ t_name }}.h>
+#include <etsi_its_{{ etsi_type }}_coding/{{ etsi_type }}_{{ asn1_type_name }}.h>
```
Comment (lreiher) [inline utils/codegen/codegen-py/templates/convertSequenceType.h.jinja2:29]: use `{{x}}` instead of `{{ x }}` for all jinja templates

Comment (jpbusch) [reply utils/codegen/codegen-py/templates/convertSequenceType.h.jinja2]: Done with cc460684aa93d203e84fbc8c32403245c3bc1605.

**Pull request merged**